types/index.d.ts

```diff
@@ -398,6 +398,7 @@ export namespace Server {
 
     interface Options {
         allowHttp: boolean;
+        headers?: Map<string, string>;
```

Comment:
It is working with axios. Added a test to make sure it is working.
I prefer to work with a map, I think it is more readable and in JS you can use both anyway.
The limitation is only for Typescript users. With map it is easier to manipulate the object because it is an existing data structure.